check_clang_tidy.sh

```diff
@@ -16,7 +16,27 @@ _travis_run_clang_tidy_fix() {
     for pkg in ${SOURCE_PKGS[@]} ; do  # process files in topological order
         test -z "${PKGS[$pkg]}" && continue  # skip pkgs without compile_commands.json
         travis_fold start clang.tidy "  - $(colorize BLUE Processing $pkg)"
-        travis_run_wait "$RUN_CLANG_TIDY_EXECUTABLE -fix -p ${PKGS[$pkg]} 2> /dev/null"
+
+        # Find all .cpp files in pkg's src_dir that were added or modified in this pull request
+        # If we are not processing a Travis pull request, check all files
+        # To enable unit-testing, check all files when $TEST_PKG is defined
+        modified_files=()
```

Comment:
This function is getting large with a lot of indentation. Could you move it into a subfunction? I think this would improve readability, I'm having a hard time following this logic

Reply:
I tried to do so in the very first place, but you cannot return an array (here `modified_files`) from a bash function.

Reply:
I now implemented a function that fills an array passed by reference.